### PR TITLE
*updated PR* npo_watchlist: Rewrite to support new npo.nl layout

### LINKED ISSUE #1892: BUG: Unhandled error in plugin npo_watchlist: 'NoneType' object has no attribute 'find'
latest version of flexget and download-npo installed.

config.yml
```yaml
tasks:
tv-journaal:
npo_watchlist:
email: xxx
password: yyy
remove_accepted: yes
max_episode_age_days: 7
accept_all: yes
exec:
fail_entries: yes
auto_escape: yes
on_output:
for_accepted:
- /usr/local/bin/download-npo -o "/var/storage/queue/tvshows/NPO/{{series_name_plain}}" -f "{serie_titel} - {datum} {aflevering_titel} ({episode_id})" -t {{url}}
```
```
2017-07-05 20:31 CRITICAL task tv-journaal BUG: Unhandled error in plugin npo_watchlist: 'NoneType' object has no attribute 'find'
Traceback (most recent call last):
File "/usr/local/lib/python2.7/dist-packages/flexget/task.py", line 486, in __run_plugin
return method(*args, **kwargs)
File "/usr/local/lib/python2.7/dist-packages/flexget/event.py", line 23, in call
return self.func(*args, **kwargs)
File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/input/npo_watchlist.py", line 326, in on_task_input
entries += self._get_favorites_entries(task, config)
File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/input/npo_watchlist.py", line 318, in _get_favorites_entries
series_info = self._get_series_info(task, config, series_name, url)
File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/input/npo_watchlist.py", line 180, in _get_series_info
series_info['npo_url'] = tvseries.find('a', itemprop='url')['href']
AttributeError: 'NoneType' object has no attribute 'find'
```
Something changed on the NPO site?

### LINKED ISSUE #1893: npo.nl site is overhauled and has new account system
This is a reminder to myself to update the npo_watchlist plugin for the new npo.nl account system for flexget.

I don't have time atm but will look into it asap

### PR DESCRIPTION
**Pull request by mfonville**
updated PR of #1901 with some minor fixes/changes and updated tests

Closes #1892 and closes #1893 and closes #1901.

### CODE REVIEW

**Comment by mfonville** [comment]
@jeroenl

**Comment by liiight** [comment]
does this replace #1901 then?

**Comment by mfonville** [comment]
yes, this supersedes #1901

**Comment by liiight** [comment]
you guys should have used the same branch then :)
closing #1901

**Comment by mfonville** [comment]
I can't commit to his personal branch ;-)

File: flexget/plugins/input/npo_watchlist.py [hunk 1]
```diff
@@ -71,259 +66,192 @@ def _convert_plain(self, s):
...
+        if 'isAuthenticatedUser' in requests.cookies:
             log.debug('Already logged in')
-            return login_response
-        elif login_response.url != 'https://mijn.npo.nl/inloggen':
+            return
+
+        login_url = 'https://www.npo.nl/login'
+        login_api_url = 'https://www.npo.nl/api/login'
+
+        login_response = requests.get(login_url)
```
Comment (liiight) [inline flexget/plugins/input/npo_watchlist.py]: Wrap this in try except

File: flexget/plugins/input/npo_watchlist.py [hunk 2]
```diff
@@ -71,259 +66,192 @@ def _convert_plain(self, s):
...
-            e['series_date'] = entry_date
-            e['series_id_type'] = 'date'
-            e['description'] = list_item.find('p').text
-            e['remove_url'] = self._prefix_url('https://mijn.npo.nl', remove_url)
+        series_url = page.find('a', class_='npo-episode-header-program-info')['href']
 
-            if config.get('remove_accepted'):
-                e.on_complete(self.entry_complete, task=task)
+        if series_url not in self._series_info:
+            response = requests.get(series_url)
```
Comment (liiight) [inline flexget/plugins/input/npo_watchlist.py]: Try except

**Comment by liiight** [comment]
A few other requests that aren't wrapped in try except. Fix those and I think we're good

**Comment by mfonville** [comment]
I now updated the branch with the try-except statements for the requests, can this please be merged so that the plugin works again? Thanks!

**Comment by mfonville** [comment]
As I wrote in Gitter, I am stuck why the Travis build keeps failing (for me I get even the same error on flexget/develop branch), I had hoped that the exception stuff should now have been fixed. If one of the devs could take a look, that'd be a great help because I will be afk for awhile.

**Comment by liiight** [comment]
@mfonville as i wrote to you on gitter, build crashes were unrelated to your code. merge from dev and it should be fine

**Pull request merged**